eslint.config.mjs

```diff
@@ -0,0 +1,366 @@
+import jest from 'eslint-plugin-jest';
+import lwcInternal from '@lwc/eslint-plugin-lwc-internal';
+import _import from 'eslint-plugin-import';
+import header from 'eslint-plugin-header';
+import { fixupPluginRules } from '@eslint/compat';
+import globals from 'globals';
+import js from '@eslint/js';
+import tseslint from 'typescript-eslint';
+
+export default [
+    {
+        ignores: [
+            '**/node_modules/',
+            '**/dist/',
+            '**/lib/',
+            '**/coverage/',
+            '**/fixtures/',
+            '**/public/',
+            '**/__benchmarks_results__/',
+            '**/playground/',
+            '**/*.html',
+            '**/*.css',
+        ],
+    },
+    js.configs.recommended,
+    ...tseslint.configs.recommendedTypeChecked,
+    {
+        files: ['**/*.ts', '**/*.mjs', '**/*.js'],
```

Comment:
```suggestion
```
The only JS/TS file excluded by this pattern is `./packages/lwc/jest.config.cjs`, so I think we can just drop it. (Unless we need it to pick up `.ts`?)

Reply:
Not sure I understood this comment.

Reply:
* There's only one JavaScript/TypeScript file in our project that's _not_ a `.ts`, `.mjs`, or `.js` file. (The file is `./packages/lwc/jest.config.js`.)
* If we remove the `files` list and `eslint .` applies these rules to `.ts` files, then we should remove the `files` list, because it doesn't accomplish anything special.
* If we remove the `files` list and `eslint .` does not apply the rules to `.ts` files, then we'll need to keep it. Oh well.